x/blsgroup/keeper/msg_server.go

```diff
@@ -0,0 +1,106 @@
+package keeper
+
+import (
+	"context"
+
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/x/group"
+	grouperrors "github.com/cosmos/cosmos-sdk/x/group/errors"
+
+	"github.com/fetchai/fetchd/crypto/keys/bls12381"
+	"github.com/fetchai/fetchd/x/blsgroup"
+)
+
+var _ blsgroup.MsgServer = Keeper{}
+
+func (k Keeper) VoteAgg(goCtx context.Context, req *blsgroup.MsgVoteAgg) (*blsgroup.MsgVoteAggResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	proposalResp, err := k.groupKeeper.Proposal(goCtx, &group.QueryProposalRequest{ProposalId: req.ProposalId})
+	if err != nil {
+		return nil, err
+	}
+	proposal := proposalResp.Proposal
+
+	// Ensure that we can still accept votes for this proposal.
+	if proposal.Status != group.PROPOSAL_STATUS_SUBMITTED {
+		return nil, sdkerrors.Wrap(grouperrors.ErrInvalid, "proposal not open for voting")
+	}
+	if proposal.VotingPeriodEnd.Before(ctx.BlockTime()) {
+		return nil, sdkerrors.Wrap(grouperrors.ErrExpired, "voting period has ended already")
+	}
+
+	policyInfo, err := k.groupKeeper.GroupPolicyInfo(goCtx, &group.QueryGroupPolicyInfoRequest{Address: proposal.Address})
+	if err != nil {
+		return nil, sdkerrors.Wrap(err, "load group policy")
+	}
+
+	electorate, err := k.groupKeeper.GroupInfo(goCtx, &group.QueryGroupInfoRequest{GroupId: policyInfo.Info.GroupId})
+	if err != nil {
+		return nil, err
+	}
+
+	groupMemberResp, err := k.groupKeeper.GroupMembers(goCtx, &group.QueryGroupMembersRequest{GroupId: electorate.Info.Id})
```

Comment:
The problem of  k.groupKeeper.GroupMembers is they use pagination and there is a default limit of 100. If we have more than 100 group members, we won't be able to get them all

Reply:
Tracked in #242 - updating is easy but the SDK has a bug and fails to serve groups with many members.